src/hdmf_zarr/utils.py

```diff
@@ -1,22 +1,129 @@
-"""Collection of utility I/O classes for the ZarrIO backend store"""
-from zarr.hierarchy import Group
-import zarr
-import numcodecs
-import numpy as np
+"""Collection of utility I/O classes for the ZarrIO backend store."""
+import gc
+import traceback
+import multiprocessing
+import math
+import json
+import logging
 from collections import deque
 from collections.abc import Iterable
+from typing import Optional, Union, Literal, Tuple
+from pathlib import Path
+from concurrent.futures import ProcessPoolExecutor
+from threadpoolctl import threadpool_limits
+from warnings import warn
 
-import json
-import logging
+import numcodecs
+import zarr
+import numpy as np
+from zarr.hierarchy import Group
 
-from hdmf.data_utils import DataIO
+from hdmf.data_utils import DataIO, GenericDataChunkIterator, DataChunkIterator, DataChunk, AbstractDataChunkIterator
+from hdmf.query import HDMFDataset
 from hdmf.utils import (docval,
                         getargs)
 
 from hdmf.spec import (SpecWriter,
                        SpecReader)
 
 
+# Necessary definitions to avoid parallelization bugs
+# Inherited from SpikeInterface experience
+
+# see
+# https://stackoverflow.com/questions/10117073/how-to-use-initializer-to-set-up-my-multiprocess-pool
+# the tricks is : theses 2 variables are global per worker
+# so they are not share in the same process
+global _worker_context
+global _operation_to_run
+
+
```

Comment:
As far as I can tell the module level functions `_write_buffer_zarr`,  `_is_pickleable`, `function_wrapper`, and `initializer_wrapper` are actually specific for `ZarrIODataChunkIteratorQueue` (i.e., they are only used there). If that is the case, I would suggest declaring them as static methods as part of the `ZarrIODataChunkIteratorQueue`  class rather than at the module level (which implies more general functionality that is not specific to a particular class).

Reply:
Sounds fair

Fixed; all non-global definitions moved attached to the Queue class as staticmethods